test/jest/acceptance/iac/file-output.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { readFileSync, unlinkSync } from 'fs';
+import * as path from 'path';
+import { v4 as uuidv4 } from 'uuid';
+import { startMockServer } from './helpers';
+
+jest.setTimeout(50000);
+
+describe('iac test --json-file-output', () => {
+  let run: (
+    cmd: string,
+  ) => Promise<{ stdout: string; stderr: string; exitCode: number }>;
+  let teardown: () => void;
+
+  beforeAll(async () => {
+    const result = await startMockServer();
+    run = result.run;
+    teardown = result.teardown;
+  });
+
+  afterAll(async () => teardown());
+
+  it('can save JSON output to file while sending human readable output to stdout', async () => {
+    const jsonOutputFilename = path.join(__dirname, `${uuidv4()}.json`);
+    const { stdout } = await run(
+      `snyk iac test ./iac/terraform/sg_open_ssh.tf --json-file-output=${jsonOutputFilename}`,
+    );
+    expect(stdout).toMatch('Organization:');
+
+    const outputFileContents = readFileSync(jsonOutputFilename, 'utf-8');
+    unlinkSync(jsonOutputFilename);
+    const jsonObj = JSON.parse(outputFileContents);
+    const okValue = jsonObj.ok as boolean;
+    expect(okValue).toBeFalsy();
+    const lineNumber = jsonObj.infrastructureAsCodeIssues[0]
+      .lineNumber as number;
+    expect(lineNumber).not.toEqual(-1);
+  });
+});
+
+describe('iac test --sarif-file-output', () => {
+  let run: (
+    cmd: string,
+  ) => Promise<{ stdout: string; stderr: string; exitCode: number }>;
+  let teardown: () => void;
+
+  beforeAll(async () => {
+    const result = await startMockServer();
+    run = result.run;
+    teardown = result.teardown;
+  });
+
+  afterAll(async () => teardown());
+
+  it('can save Sarif output to file while sending human readable output to stdout', async () => {
+    const sarifOutputFilename = path.join(__dirname, `${uuidv4()}.sarif`);
+    const { stdout } = await run(
+      `snyk iac test ./iac/terraform/sg_open_ssh.tf --sarif-file-output=${sarifOutputFilename}`,
+    );
```

Comment:
Weird suggestion: If we wanted to avoid the tmp file write here (and these tests only run on unix) we can cheat here and have the sarif printed to stderr and read that directly:

```suggestion
    // stdout = regular output, stderr expects sarif output
    const { stdout, stderr } = await run(
      `snyk iac test ./iac/terraform/sg_open_ssh.tf --sarif-file-output=/dev/stderr`,
    );
```

Reply:
I don't see why not! I didn't think of this, good thinking

Reply:
Tried this but the tests started failing in CircleCI and not locally. I couldn't recreate it locally at all so I just swapped back to using the tmp file